src/webapps/azext_webapps/custom.py

```diff
@@ -0,0 +1,144 @@
+
+    # ResourceGroup: check if default RG is set
+    default_rg = cmd.cli_ctx.config.get('defaults', 'group')
+    if (default_rg and check_resource_group_supports_linux(cmd, default_rg, location)):
+        rg = default_rg
+        rg_mssg = "[Using default ResourceGroup]"
+    else:
+        rg_mssg = ""
+
+    runtime_version = "{}|{}".format(language, version_used_create)
+    src_path = "{} {}".format(src_dir.replace("\\", "\\\\"), str_no_contents_warn)
+    rg_str = "{} {}".format(rg, rg_mssg)
+
+    dry_run_str = r""" {
+            "name" : "%s",
+            "appServicePlan" : "%s",
+            "resourceGroup" : "%s",
+            "sku": "%s",
+            "os": "%s",
+            "location" : "%s",
+            "src_path" : "%s",
+            "version_detected": "%s",
+            "version_to_create": "%s"
+            }
+            """ % (name, asp, rg_str, full_sku, os_val, location, src_path,
+                   node_version, runtime_version)
+
+    create_json = json.dumps(json.loads(dry_run_str), indent=4, sort_keys=True)
+    if dryrun:
+        logger.warning("""
+            Web app will be created with the below configuration,
+            re-run command without the --dryrun flag to create & deploy a new app
+            """)
+        return logger.warning(create_json)
+
+    # create RG if the RG doesn't already exist
+    if not check_resource_group_exists(cmd, rg):
+        logger.warning("Creating ResourceGroup '%s' ...", rg)
+        create_resource_group(cmd, rg, location)
+        logger.warning("ResourceGroup creation complete")
+    else:
+        logger.warning("Resource Group '%s' already exists.", rg)
+
+    # create asp
+    if not check_if_asp_exists(cmd, rg, asp):
+        logger.warning("Creating App service plan '%s' ...", asp)
+        sku_def = SkuDescription(tier=full_sku, name=sku, capacity=1)
+        plan_def = AppServicePlan(loc_name, app_service_plan_name=asp,
+                                  sku=sku_def, reserved=True)
+        client.app_service_plans.create_or_update(rg, asp, plan_def)
+        logger.warning("AppServicePlan creation complete")
+    else:
+        logger.warning("AppServicePlan '%s' already exists.", asp)
+
+    # create the Linux app
+    logger.warning("Creating app '%s' ....", name)
+    create_webapp(cmd, rg, name, asp, runtime_version)
+    logger.warning("Webapp creation complete")
+
+    # setting to build after deployment
+    logger.warning("Updating app settings to enable build after deployment")
+    update_app_settings(cmd, rg, name, ["SCM_DO_BUILD_DURING_DEPLOYMENT=true"])
+
+    # zip contents & deploy
+    logger.warning("Creating zip with contents of dir %s ...", src_dir)
+    zip_file_path = zip_contents_from_dir(src_dir)
+
+    logger.warning("Deploying and building contents to app. This operation can take some time to finish...")
+    enable_zip_deploy(cmd, rg, name, zip_file_path)
+    return logger.warning("All done. %s", create_json)
```

Comment:
You don't need to return with logger.warning.
Just leave it and by default Python will return None or explicitly add `return None` at the end.